### PR TITLE
Refactor the 'data' property of the 'Code instumentaion' subsystem

### PR DESCRIPTION
**Pull request by LavrovArtem**
part of #1379

### CODE REVIEW

**Comment by LavrovArtem** [comment]
@testcafe-build-bot \retest

File: src/client/sandbox/code-instrumentation/properties/index.js [hunk 1]
```diff
@@ -192,24 +192,11 @@ export default class PropertyAccessorsInstrumentation extends SandboxBase {
...
-                    return el.data.message;
-                },
-
-                set: (el, value) => {
-                    if (domUtils.isDomElement(el))
-                        this.elementSandbox.setAttributeCore(el, ['data', value]);
-
-                    return value;
-                }
+                condition: evt => PropertyAccessorsInstrumentation._isSupportedMessageEvent(evt),
```
Comment (miherlosev) [inline src/client/sandbox/code-instrumentation/properties/index.js]: Let's rename to `PropertyAccessorsInstrumentation._isMessageEventWithoutDataPropGetter(evt)`

File: src/client/sandbox/native-methods.js [hunk 2]
```diff
@@ -175,15 +175,12 @@ class NativeMethods {
...
-        if (inputValueDescriptor && typeof inputValueDescriptor.set === 'function')
-            this.inputValueSetter = inputValueDescriptor.set;
+        const objectDataDescriptor = win.Object.getOwnPropertyDescriptor(win.HTMLObjectElement.prototype, 'data');
 
-        if (textAreaValueDescriptor && typeof textAreaValueDescriptor.set === 'function')
-            this.textAreaValueSetter = textAreaValueDescriptor.set;
+        // Setters
+        this.inputValueSetter    = win.Object.getOwnPropertyDescriptor(win.HTMLInputElement.prototype, 'value').set;
+        this.textAreaValueSetter = win.Object.getOwnPropertyDescriptor(win.HTMLTextAreaElement.prototype, 'value').set;
+        this.objectDataSetter    = objectDataDescriptor.set;
```
Comment (miherlosev) [inline src/client/sandbox/native-methods.js]: Why object data getter definition
`this.objectDataSetter    = objectDataDescriptor.set;`
placed not on the next line after
`this.objectDataGetter               = objectDataDescriptor.get;`
?

File: test/client/fixtures/sandbox/node/attributes-test.js [hunk 3]
```diff
@@ -122,8 +120,7 @@ test('url', function () {
         { tagName: 'script', attr: 'src' },
         { tagName: 'link', attr: 'href' },
         { tagName: 'form', attr: 'action' },
-        { tagName: 'embed', attr: 'src' },
-        { tagName: 'object', attr: 'data' }
+        { tagName: 'embed', attr: 'src' }
     ];
 
     if (!browserUtils.isIE || browserUtils.version > 9) {
```
Comment (miherlosev) [inline test/client/fixtures/sandbox/node/attributes-test.js]: Remove this condition here and in all places (It is for IE9+)

File: src/client/sandbox/native-methods.js (hunk 2 above)
Comment (LavrovArtem) [reply src/client/sandbox/native-methods.js]: Because functions grouped by setters and getters.

Comment (miherlosev) [reply src/client/sandbox/native-methods.js]: Ok

**Comment by LavrovArtem** [comment]
FPR

File: test/client/fixtures/sandbox/node/attributes-test.js (hunk 3 above)
Comment (miherlosev) [reply test/client/fixtures/sandbox/node/attributes-test.js]: What about it?

Comment (LavrovArtem) [reply test/client/fixtures/sandbox/node/attributes-test.js]: https://github.com/DevExpress/testcafe-hammerhead/pull/1479/files#diff-4b08ab790386be4d04295aa29c8be49eL129

Comment (miherlosev) [reply test/client/fixtures/sandbox/node/attributes-test.js]: Ok

**Comment by LavrovArtem** [comment]
ping @churkin

**Pull request merged**